src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisConfigForm.tsx

```diff
@@ -35,118 +35,139 @@
 import * as React from 'react';
 import { useTranslation } from 'react-i18next';
-import { concatMap, filter, first, Observable, take } from 'rxjs';
+import { first, ReplaySubject, take } from 'rxjs';
 
 interface AutomatedAnalysisConfigFormProps {
   useTitle?: boolean;
-  targetObs?: Observable<Target>;
+  inlineForm?: boolean;
+}
+
+interface FormConfig {
+  maxAge: number;
+  maxAgeUnits: number;
+  maxSize: number;
+  maxSizeUnits: number;
+  template: Pick<Partial<EventTemplate>, 'name' | 'type'>;
 }
 
 export const AutomatedAnalysisConfigForm: React.FC<AutomatedAnalysisConfigFormProps> = ({
   useTitle = false,
-  targetObs,
+  inlineForm = false,
 }) => {
   const context = React.useContext(ServiceContext);
   const addSubscription = useSubscriptions();
   const { t } = useTranslation();
 
-  const parseEventString = React.useMemo((): Pick<Partial<EventTemplate>, 'name' | 'type'> => {
-    const eventString = context.settings.automatedAnalysisRecordingConfig().template;
-    if (!eventString) {
-      return {};
-    }
-    const templateName = eventString.split(',')[0].split('=')[1];
-    const templateType = eventString.split(',')[1].split('=')[1];
-    if (!(templateType === 'TARGET' || templateType === 'CUSTOM')) {
-      console.error(`Invalid template type ${templateType}`);
-      return {};
-    }
-    return {
-      name: templateName,
-      type: templateType,
-    };
-  }, [context.settings]);
+  const targetSubjectRef = React.useRef(new ReplaySubject<Target>(1));
+  const targetSubject = targetSubjectRef.current;
 
   const [recordingConfig, setRecordingConfig] = React.useState<AutomatedAnalysisRecordingConfig>(
     context.settings.automatedAnalysisRecordingConfig()
   );
+  const [formConfig, setFormConfig] = React.useState<FormConfig>({
+    maxAge: context.settings.automatedAnalysisRecordingConfig().maxAge,
+    maxAgeUnits: 1,
+    maxSize: context.settings.automatedAnalysisRecordingConfig().maxSize,
+    maxSizeUnits: 1,
+    template: context.settings.automatedAnalysisRecordingConfig().template,
+  });
 
   const [templates, setTemplates] = React.useState<EventTemplate[]>([]);
-  const [template, setTemplate] = React.useState<Pick<Partial<EventTemplate>, 'name' | 'type'>>(parseEventString);
-  const [maxAge, setMaxAge] = React.useState(recordingConfig.maxAge);
-  const [maxAgeUnits, setMaxAgeUnits] = React.useState(1);
-  const [maxSize, setMaxSize] = React.useState(recordingConfig.maxSize);
-  const [maxSizeUnits, setMaxSizeUnits] = React.useState(1);
   const [errorMessage, setErrorMessage] = React.useState('');
-  const [isLoading, setIsLoading] = React.useState(true);
-
-  const refreshTemplates = React.useCallback(() => {
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [editing, setEditing] = React.useState(false);
+  const [isAuthModalOpen, setIsAuthModalOpen] = React.useState(false);
+  const refreshTemplates = React.useCallback((target: Target) => {
+    if (target === NO_TARGET) {
```

Comment:
Actually, on another thoughts, we should have `NO_TARGET` case.

How about using `iif` operator to check `target === NO_TARGET` that returns `[]` when fetching the templates.